### ISSUE TITLE
Four random box

### ISSUE THREAD

**Issue by Darkhan92** [reiniscimurs/DRL-robot-navigation#44]
Hello, dear Reinis Cimurs,

 First of all, thank you for publishing such great work. I want to use this model for our own custom robot as a base.

Here are some problems I have encountered while running the simulation. When I run it, it works fairly well, it has to circle around the problem but I have solved it by changing the seed number. But another issue is it shows error that it cannot load four cardboard boxes, saying this box does not exist. So Rvis and Gazebo loading with plain empty space, without boxes as they should be. What is the problem can be for that, in python code I did not found where happens this problem while I saw where we define the random box function. 
![image](https://user-images.githubusercontent.com/49669134/211811134-c2fc2e8c-4fe5-48ca-9758-21bfd3849b1f.png)

Here is where it is defined but still I did not see where it is defined walls and boxes as in picture

![image](https://user-images.githubusercontent.com/49669134/211811433-47aa000f-0c83-4f2e-9d05-a8565fbee4e8.png)

![image](https://user-images.githubusercontent.com/49669134/211811694-005ff336-576e-4fa5-8416-c61fa718a540.png)

![image](https://user-images.githubusercontent.com/49669134/211811783-5a88e208-3789-4a36-bd94-8fb841145836.png)

**Comment by reiniscimurs** [comment]
Hi,

Which repo version (branch) did you use? The issue is that the correct world environment is not loaded. The environment should not be empty and should have obstacles and boxes in it. This could happen if something is not sourced properly.

Let me know the full terminal output when you start the training script. Also did you run the full sourcing commands as per the tutorial?

**Comment by reiniscimurs** [comment]
This repo currently has 3 branches, all are slightly different. When I asked about the version, I meant which branch was used as sometimes there can be discripancies there.

Check if answers in this issue would not solve your problem: https://github.com/reiniscimurs/DRL-robot-navigation/issues/30
at might be missing).
Usually it would be enough to just install all the missing packages to your base environment. If ros packages is missing, then install them with pip3 and the same for numpy (or any other relevant packages that might be missing). If using virtual environment, you will have to make sure you are sourcing the locations of the repo, where to look for the files.

**Comment by reiniscimurs** [comment]
Looks like you are exporting the wrong gazebo resource path. These are absolute paths. It looks like your repo is stored in ~/Desktop/MOB_robot_RL
So your gazebo resource path export command should be: `GAZEBO_RESOURCE_PATH= ~/Desktop/MOB_robot_RL/DRL-robot-navigation/catkin_ws/src/multi_robot_scenario/launch`

Try it out and see if it works.

This issue is explained in https://github.com/reiniscimurs/DRL-robot-navigation/issues/30 as well as setting up the paths is explained in the tutorial: https://medium.com/p/d62715722303

**Comment by reiniscimurs** [comment]
Did you try with `export GAZEBO_RESOURCE_PATH= ~/Desktop/MOB_robot_RL/DRL-robot-navigation/catkin_ws/src/multi_robot_scenario/launch` ?

**Comment by reiniscimurs** [comment]
I can easily reproduce this issue by not setting or setting the wrong GAZEBO_RESOURCE_PATH which leads me to believe that the path to your repo is not set up quite right.

What you can do is to check GAZEBO_RESOURCE_PATH value before and after you execute the export command.

1. Open a new terminal and execute `echo $GAZEBO_RESOURCE_PATH` and note the result
2. Execute the other export commands :
`export ROS_HOSTNAME=localhost`
`export ROS_MASTER_URI=http://localhost:11311`
`export ROS_PORT_SIM=11311`
`export GAZEBO_RESOURCE_PATH= ~/Desktop/MOB_robot_RL/DRL-robot-navigation/catkin_ws/src/multi_robot_scenario/launch`
`source ~/.bashrc`
`source devel/setup.bash`

3. Then execute the `echo $GAZEBO_RESOURCE_PATH` and note what the output is again

**Comment by reiniscimurs** [comment]
In the Gazebo resource path you can see you have repeating /home/issai part. That is what seems to be causing the issue.This happens because you are exporting `~/home/issai/Desktop/MOB_robot_RL/DRL-robot-navigation/catkin_ws/src/multi_robot_scenario/launch`
The `~` sign already specifies home and user and is equal to `home/issai/`
So what you end up actually exporting is `/home/issai/home/issai/Desktop/MOB_robot_RL/DRL-robot-navigation/catkin_ws/src/multi_robot_scenario/launch`

So you should export either `/home/issai/Desktop/MOB_robot_RL/DRL-robot-navigation/catkin_ws/src/multi_robot_scenario/launch`
or
`~/Desktop/MOB_robot_RL/DRL-robot-navigation/catkin_ws/src/multi_robot_scenario/launch` 
with the ~ included

See what the echo $GAZEBO_RESOURCE_PATH says about these exported paths and whether it is equal to the path you need

**Comment by reiniscimurs** [comment]
Great that you managed to get it working!

**Comment by reiniscimurs** [comment]
I would assume a larger gamma might work better, as you would have a longer future window that would impact each decision. To me the default hyperparameters in this repo work quite fine, but there is some randomness involved so you might want to try out setting different seed values. Generally, you should start seeing collision avoidance behavior between 20 to 40 epochs.


If you have a working robot model, it should not be too difficult to change the robot. For depth images though, be vary of the field of view as a single camera tends not to pick up obstacles in robots periphery.
